docs/reference/buildx_ls.md

```diff
@@ -7,23 +7,66 @@ docker buildx ls
 <!---MARKER_GEN_START-->
 List builder instances
 
+### Options
+
+| Name                  | Type     | Default | Description       |
+|:----------------------|:---------|:--------|:------------------|
+| [`--format`](#format) | `string` | `table` | Format the output |
+
 
 <!---MARKER_GEN_END-->
 
 ## Description
 
-Lists all builder instances and the nodes for each instance
+Lists all builder instances and the nodes for each instance.
 
 ```console
 $ docker buildx ls
-NAME/NODE       DRIVER/ENDPOINT             STATUS  BUILDKIT PLATFORMS
-elated_tesla *  docker-container
-  elated_tesla0 unix:///var/run/docker.sock running v0.10.3  linux/amd64
-  elated_tesla1 ssh://ubuntu@1.2.3.4        running v0.10.3  linux/arm64*, linux/arm/v7, linux/arm/v6
-default         docker
-  default       default                     running v0.8.2   linux/amd64
+NAME/NODE           DRIVER/ENDPOINT                   STATUS    BUILDKIT   PLATFORMS
+elated_tesla*       docker-container
+ \_ elated_tesla0    \_ unix:///var/run/docker.sock   running   v0.10.3    linux/amd64
+ \_ elated_tesla1    \_ ssh://ubuntu@1.2.3.4          running   v0.10.3    linux/arm64*, linux/arm/v7, linux/arm/v6
+default             docker
+ \_ default          \_ default                       running   v0.8.2     linux/amd64
 ```
 
 Each builder has one or more nodes associated with it. The current builder's
 name is marked with a `*` in `NAME/NODE` and explicit node to build against for
 the target platform marked with a `*` in the `PLATFORMS` column.
+
+## Examples
+
+### <a name="format"></a> Format the output (--format)
+
+The formatting options (`--format`) pretty-prints builder instances output
+using a Go template.
+
+Valid placeholders for the Go template are listed below:
+
+| Placeholder       | Description                                 |
+|-------------------|---------------------------------------------|
+| `.NameNode`       | Name of the builder or node                 |
+| `.DriverEndpoint` | Driver (for builder) or Endpoint (for node) |
+| `.LastActivity`   | Builder last activity                       |
+| `.Status`         | Builder or node status                      |
+| `.Buildkit`       | BuildKit version of the node                |
+| `.Platforms`      | Available node's platforms                  |
+| `.Error`          | Error                                       |
+| `.IsNode`         | `true` if entry is a node                   |
+| `.IsCurrent`      | `true` if entry is the active builder       |
```

Comment:
Hm. I'm not sure I'm a fan of this approach for writing templates, I think we lose something from the structured data.

I think I'd rather have `.` be the builder, and have a `.Nodes` property contain a list of nodes that can be `range`d or `index`d to get the node details out.

Reply:
Yeah me too but that's how the cli api does it today using the formatter context. I don't think there is something else than json-lines in the cli atm (cc @thaJeztah)

Reply:
Ah I see @thaJeztah commented on this https://github.com/docker/buildx/pull/1787#pullrequestreview-1422131042. I would also prefer `json-array` tbh.

Reply:
@jedevc was that specific about the `.IsCurrent()` ? I know we added that because in the past we had `*` as part of the name, which was not ideal, and having a separate `.IsCurrent()` allow it to be "formatted how you'd like", but still keeping the raw name available to be used.

Reply:
Ah sorry my bad, comment on the wrong line :cry: I meant to comment on `IsNode`.

Reply:
I was looking at https://docs.docker.com/engine/reference/commandline/service_ps/ which services and tasks (similar to us with builders and nodes) and it seems format only handles tasks and not services: https://docs.docker.com/engine/reference/commandline/service_ps/#format